provider/resource_keycloak_openid_client_service_account_role.go

```diff
@@ -40,18 +40,32 @@ func resourceKeycloakOpenidClientServiceAccountRole() *schema.Resource {
 	}
 }
 
-func getOpenidClientServiceAccountRoleFromData(data *schema.ResourceData) *keycloak.OpenidClientServiceAccountRole {
-	return &keycloak.OpenidClientServiceAccountRole{
-		Id:                   data.Id(),
-		ContainerId:          data.Get("client_id").(string),
-		Name:                 data.Get("role").(string),
-		RealmId:              data.Get("realm_id").(string),
-		ServiceAccountUserId: data.Get("service_account_user_id").(string),
+func getOpenidClientServiceAccountRoleFromData(data *schema.ResourceData, keycloakClient *keycloak.KeycloakClient) (*keycloak.OpenidClientServiceAccountRole, error) {
+	containerId := data.Get("client_id").(string)
+	roleName := data.Get("role").(string)
+	realmId := data.Get("realm_id").(string)
+	serviceAccountRoleId := data.Get("service_account_user_id").(string)
+
+	role, err := keycloakClient.GetRoleByName(realmId, containerId, roleName)
+	if err != nil {
+		if handleNotFoundError(err, data) == nil {
```

Comment:
Okay, your logic makes sense, thanks for the explanation.  I think, however, that all you really want to do is check if the error is a 404 and not necessarily remove the resource from state.
```suggestion
		if keycloak.ErrorIs404(err) {
```

Reply:
yes :)

Reply:
@mrparkers just updated